realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java

```diff
@@ -51,8 +51,93 @@
+ *
+ * The <i>App</i> can be used to:
+ * <ul>
+ *   <li>Register uses and perform various user related operations through authentication providers
+ *   ({@link io.realm.mongodb.auth.ApiKeyAuth}, {@link EmailPasswordAuthImpl})</li>
+ *   <li>Synchronize data between the local device and a remote Realm App with Synchronized Realms</li>
+ *   <li>Invoke Realm App functions with {@link Functions}</li>
+ *   <li>Access remote data from MongoDB databases with a {@link io.realm.mongodb.mongo.MongoClient}</li>
+ * </ul>
+ * <p>
+ * To create an app that is linked with a remote <i>Realm App</i> initialize Realm and configure the
+ * <i>App</i> as below:
+ * <p>
+ * <pre>
+ *    class MyApplication extends Application {
+ *
+ *         @Override
+ *         public void onCreate() {
+ *             super.onCreate();
+ *
+ *             Realm.init(this);
+ *
+ *             AppConfiguration appConfiguration = new AppConfiguration.Builder(BuildConfig.MONGODB_REALM_APP_ID)
+ *                     .appName(BuildConfig.VERSION_NAME)
+ *                     .appVersion(Integer.toString(BuildConfig.VERSION_CODE))
+ *                     .build();
+ *
+ *             App APP = new App(appConfiguration);
+ *         }
+ *
+ *     }
+ * </pre>
+ * <p>
+ * After configuring the <i>App</i> you can start managing users, configure Synchronized Realms,
+ * call remote Realm Functions and access remote data through Mongo Collections. The below examples
+ * shows the synchronized APIs which cannot be used from the main thread. For the equivalent
+ * asynchronous counter parts. The example project in please see
+ * https://github.com/realm/realm-java/tree/v10/examples/mongoDbRealmExample.
+ *
+ * To register a new user and/or login with an existing user as below:
+ * <pre>
+ *     // Register new user
+ *     User user = APP.getEmailPasswordAuth().registerUser(username, password);
+ *
+ *     // Login with existing
+ *     APP.login(Credentials.emailPassword(username, password))
+ * </pre>
+ * <p>
+ * With an authorized user you can synchronize data between the local device and the remote Realm
+ * App by opening a Realm with a {@link io.realm.mongodb.sync.SyncConfiguration} as below:
+ * <pre>
+ *     SyncConfiguration syncConfiguration = new SyncConfiguration.Builder(user, "<partition value>")
+ *              .build();
+ *
+ *     Realm instance = Realm.getInstance(syncConfiguration);
+ *     SyncSession session = APP.getSync().getSession(syncConfiguration);
+ *
+ *     instance.executeTransaction(realm -> {
+ *         realm.insert(...);
+ *     });
+ *     session.uploadAllLocalChanges();
+ *     instance.close();
+ * </pre>
+ * <p>
+ * You can call remove Realm functions as below:
+ * <pre>
+ *     Functions functions = user.getFunctions();
+ *     Integer sum = functions.callFunction("sum", Arrays.asList(1, 2, 3, 4), Integer.class);
+ * </pre>
+ * <p>
+ * And access collections from the remote Realm App as below:
```

Comment:
Trying not to abuse the `shown/indicate below`:
```suggestion
 * And access collections from the remote Realm App as shown here:
```